staff/asix/full-task-app-1.0.0/taskList-api/taskRouter.js

```diff
@@ -0,0 +1,111 @@
+require('dotenv').config()
+
+const express = require('express')
+const bodyParser = require('body-parser')
+
+const app = express()
+
+let taskslist = []
+const jsonBodyParser = bodyParser.json()
+
+app.get('/api/tasks', (req,res) => {
+    res.json(taskslist)
+})
+
+app.post('/api/tasks', jsonBodyParser, (req,res) => {
+    const { text } = req.body
+    let index
+
+    if(!text) res.json(ko('Task registration failed.','You dont defined task'))
+
+    if((taskslist.length) === 0){
+        index = 1;
+    }
+    else{
+        index = taskslist[taskslist.length -1].id + 1
+    }
+
+    taskslist.push({id:index,text,done:false})
+
+    res.json(ok('Task registration succeeded.'))
+})
+
+app.put('/api/tasks/:id', (req,res) => {
+    const { id } = req.params
+    
+    if(!id) res.json(ko('Task update failed.','Invalid id'))
```

Comment:
you can put this in another layer ---> taskLogic.js maybe

Reply:
Agreed with Robert.